packages/shorebird_cli/lib/src/commands/patch_command.dart

```diff
@@ -310,8 +324,42 @@ https://github.com/shorebirdtech/shorebird/issues/472
         return ExitCode.software.code;
       }
     }
+
+    final String releaseAabPath;
+    try {
+      releaseAabPath = await _downloadReleaseArtifact(
+        Uri.parse(releaseAabArtifact.url),
+      );
+    } catch (error) {
+      downloadReleaseArtifactProgress.fail('$error');
+      return ExitCode.software.code;
+    }
+
     downloadReleaseArtifactProgress.complete();
 
+    final contentDiffs = _aabDiffer.aabContentDifferences(
+      releaseAabPath,
+      bundlePath,
+    );
+
+    logger.detail('content diffs detected: $contentDiffs');
+
+    if (contentDiffs.contains(AabDifferences.native)) {
+      logger.err(
+        '''Your aab contains native changes, which cannot be applied in a patch. Please create a new release or revert these changes.''',
+      );
+      return ExitCode.software.code;
+    }
+
+    if (contentDiffs.contains(AabDifferences.assets)) {
+      final shouldContinue = logger.confirm(
+        '''Your aab contains asset changes, which will not be included in the patch. Continue anyways?''',
```

Comment:
Can we list the asset changes?

Reply:
This seems doable, but there is some subtlety to it. Types of asset changes that will need to be handled differently:

1. Adding an asset (base/assets/flutter_assets/AssetManifest.json changes)
2. Changing an existing asset (the asset file has a different hash)
3. Changing a non-Flutter asset (a layout.xml or a .png that Flutter is not aware of)

I think this would be useful but I'd opt to do this in a followup PR.

Reply:
Yeah it would be super useful imo and I'm fine with it being a follow-up PR 👍

Reply:
I might say "The Android App Bundle" instead of "Your aab" 🤷

Reply:
Done.

Reply:
And filed https://github.com/shorebirdtech/shorebird/issues/542